lib/expression/eval.rb

```diff
@@ -7,11 +7,13 @@ class Eval < Parslet::Transform
 
     # in Ruby 0 is 'truthy' but that's not what most people expect
     rule(:or => { :left => subtree(:left), :right => subtree(:right) }) do
-      left == 0 ? right : (left or right)
+      result = left == 0 ? right : (left or right)
+      result.nil? ? 0 : result
```

Comment:
if we OR two nil values, don't we want the result to be nil...?  this seems to transform it into 0. not sure why we would do that